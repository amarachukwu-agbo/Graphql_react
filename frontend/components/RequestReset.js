import React, { Component } from 'react';
import gql from 'graphql-tag';
import { Mutation } from 'react-apollo';
import Form from './styles/Form';
import Error from '../components/ErrorMessage';

export const REQUEST_RESET_MUTATION = gql`
  mutation REQUEST_RESET_MUTATION(
    $email: String!,
    ) {
      requestPasswordReset(
      email: $email,
      ) {
      message
    }
  }
`;
export default class RequestReset extends Component {
  state = {
    email: ''
  };

  onChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  handleSubmit = async (e, requestReset) => {
    e.preventDefault();
    await requestReset();
    this.setState({
      email: '',
    })

  }
  render() {
    const { email } = this.state;
    return (
      <Mutation
        mutation={REQUEST_RESET_MUTATION}
        variables={this.state}
      >
        {(requestPasswordReset, { error, loading, called }) => (
          <Form method="post" onSubmit={(e) => this.handleSubmit(e, requestPasswordReset)} >
            <Error error={error} />
            {!loading && !error && called && <p>Success! Check your email for a reset link</p> }
            <fieldset disabled={loading} aria-busy={loading}>
            <h2>Request Password Reset</h2>
              <label htmlFor="email">
                Email
                <input
                  name="email"
                  type="email"
                  value={email}
                  onChange={this.onChange}
                />
              </label>
              <input type="submit" value="Request Reset"/>
            </fieldset>
          </Form>
        )}
      </Mutation>
    )
  }
}
