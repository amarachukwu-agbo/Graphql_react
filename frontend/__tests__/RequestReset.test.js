import { mount } from 'enzyme';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import RequestReset, { REQUEST_RESET_MUTATION } from '../components/RequestReset';

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: {email: 'test@gmail.com'}
    },
    result: {
      data: {
        requestPasswordReset: {
          __typename: 'message',
          message: 'Request token sent successfully'
        }
      }
    }
  }
];

describe('<RequestReset />', () => {
  it('renders correctly and matches snapshot', async () => {
    const wrapper = mount(<MockedProvider mocks={mocks}>
      <RequestReset />
    </MockedProvider>);
    expect(wrapper.find('Form')).toMatchSnapshot();
  });

  it('calls the mutation when form is submitted', async () => {
    const wrapper = mount(<MockedProvider mocks={mocks}>
      <RequestReset />
    </MockedProvider>);
    wrapper.find('input').first().simulate('change', {
      target: {
        value: 'test@gmail.com',
        name: "email"
      }
    });
    wrapper.find('form').simulate('submit');
    await wait(0);
    wrapper.update();
    expect(wrapper.find('p').text()).toContain('Success! Check your email for a reset link');
  });
});
