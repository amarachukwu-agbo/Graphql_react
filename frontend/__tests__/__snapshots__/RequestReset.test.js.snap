// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset /> renders correctly and matches snapshot 1`] = `
<Form
  method="post"
  onSubmit={[Function]}
>
  <form
    className="Form-sc-1xszr8q-0 krRxky"
    method="post"
    onSubmit={[Function]}
  >
    <DisplayError
      error={Object {}}
    />
    <fieldset
      aria-busy={false}
      disabled={false}
    >
      <h2>
        Request Password Reset
      </h2>
      <label
        htmlFor="email"
      >
        Email
        <input
          name="email"
          onChange={[Function]}
          type="email"
          value=""
        />
      </label>
      <input
        type="submit"
        value="Request Reset"
      />
    </fieldset>
  </form>
</Form>
`;
